src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -579,6 +580,19 @@ public unsafe AssemblyInfo(MonoProxy monoProxy, SessionId sessionId, string url,
             Populate();
         }
 
+        public async Task<int> DebugId(MonoSDBHelper sdbAgent, CancellationToken token)
+        {
+            if (debugId != -1)
+                return debugId;
+            debugId = await sdbAgent.GetAssemblyId(Name, token);
+            return debugId;
+        }
+
+        public void SetDebugId(int id)
+        {
+            debugId = id;
```

Comment:
Should this throw if `debugId != -1 && debugId != id` to avoid reseting the id by mistake?

Reply:
I don't like the idea to throw an exception but I added a if to set only `if (debugId == -1 && debugId != id)`